util/predicates/cluster_predicates.go

```diff
@@ -167,6 +168,45 @@ func ClusterUnpaused(logger logr.Logger) predicate.Funcs {
 	return Any(log, ClusterCreateNotPaused(log), ClusterUpdateUnpaused(log))
 }
 
+// ClusterControlPlaneInitialized returns a Predicate that returns true on Update events
+// when cluster gets ControlPlaneInitializedCondition set to true.
+// Example use:
+//  err := controller.Watch(
+//      &source.Kind{Type: &clusterv1.Cluster{}},
+//      &handler.EnqueueRequestsFromMapFunc{
+//          ToRequests: clusterToMachines,
+//      },
+//      predicates.ClusterControlPlaneInitialized(r.Log),
+//  )
+func ClusterControlPlaneInitialized(logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log := logger.WithValues("predicate", "ClusterUpdateUnpaused", "eventType", "update")
+
+			oldCluster, ok := e.ObjectOld.(*clusterv1.Cluster)
+			if !ok {
+				log.V(4).Info("Expected Cluster", "type", fmt.Sprintf("%T", e.ObjectOld))
+				return false
+			}
+			log = log.WithValues("namespace", oldCluster.Namespace, "cluster", oldCluster.Name)
+
+			newCluster := e.ObjectNew.(*clusterv1.Cluster)
+
+			if !conditions.IsTrue(oldCluster, clusterv1.ControlPlaneInitializedCondition) &&
```

Comment:
Should we take into account if the old cluster is not initializer OR the condition doesn't exist?

Reply:
Here I don't distinguish the cases. If the condition is not set or is false `conditions.IsTrue` will return false.
We only care here about the case when the condition is _added_ and it works regardless of the state of the old cluster.